rpcs3/Input/ds4_pad_handler.cpp

```diff
@@ -906,6 +922,28 @@ void ds4_pad_handler::apply_pad_data(const std::shared_ptr<PadDevice>& device, c
 		newBlinkData = true;
 	}
 
+	// Use LEDs to indicate battery level
+	// TODO: Implement a setting to enable this. The following code works and correctly indicates the current battery level by changing the LED color.
+	if (false) 
+	{
+		// TODO: Move this elsewhere. Maybe even allow custom gradients?
+		static const std::array<u32, 11> battery_level_clr = {0xff00, 0xff33, 0xff66, 0xff99, 0xffcc, 0xffff, 0xccff, 0x99ff, 0x66ff, 0x33ff, 0x00ff}; 
+		static u32 last_battery_level = 11;
+		// This makes sure that the LED color doesn't update every 1ms. DS4 only reports battery level in 10% increments
+		if (last_battery_level != ds4_dev->batteryLevel) 
+		{
+			last_battery_level = ds4_dev->batteryLevel;
+			u32 combined_color = 0xff00;
+			if (last_battery_level >= 0 && last_battery_level <= 10)
```

Comment:
since last_battery_level is unsigned it cannot be smaller than 0.
```suggestion
			if (last_battery_level < battery_level_clr.size())
```